test/lint/lint-python-dead-code.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Find dead Python code.
+
+export LC_ALL=C
+
+if ! command -v vulture > /dev/null; then
+    echo "Skipping Python dead code linting since vulture is not installed. Install by running \"pip3 install vulture\""
+    exit 0
+fi
+
+if ! vulture \
+    --min-confidence 100 \
```

Comment:
Maybe add a comment to say that this must stay at 100 to avoid false positives, which we can't deal with in this repo?

Reply:
added note as requested